awx/main/tasks.py

```diff
@@ -1482,38 +1479,85 @@ def run(self, pk, **kwargs):
-                    )
 
-                ansible_runner.utils.dump_artifacts(params)
-                isolated_manager_instance = isolated_manager.IsolatedManager(
-                    self.event_handler,
-                    canceled_callback=lambda: self.update_model(self.instance.pk).cancel_flag,
-                    check_callback=self.check_handler,
-                    pod_manager=pod_manager
-                )
-                status, rc = isolated_manager_instance.run(self.instance,
-                                                           private_data_dir,
-                                                           params.get('playbook'),
-                                                           params.get('module'),
-                                                           module_args,
-                                                           ident=str(self.instance.pk))
-                self.finished_callback(None)
+            if not isinstance(self.instance, ProjectUpdate):
+                worktype='worker'
+                # TODO: container group jobs will not work with container isolation settings
+                # but both will run with same settings when worker_in and worker_out are added
+                params['settings'].update(execution_environment_params)
             else:
-                res = ansible_runner.interface.run(**params)
-                status = res.status
-                rc = res.rc
+                worktype='worker'
+                params['settings'].update(execution_environment_params)
+
+            # Create a socketpair. Where the left side will be used for writing our payload
+            # (private data dir, kwargs). The right side will be passed to Receptor for
+            # reading.
+            sockin, sockout = socket.socketpair()
+
+            # Spawned in a thread so Receptor can start reading before we finish writing, we
+            # write our payload to the left side of our socketpair.
+            def transmit(_socket):
+                ansible_runner.interface.run(streamer='transmit',
+                                             _output=_socket.makefile('wb'),
+                                             **params)
+
+                # Socket must be shutdown here, or the reader will hang forever.
+                _socket.shutdown(socket.SHUT_WR)
+
+            threading.Thread(target=transmit, args=[sockin]).start()
+
+            # We establish a connection to the Receptor socket and submit our work, passing
+            # in the right side of our socketpair for reading.
+            receptor_ctl = ReceptorControl('/var/run/receptor/receptor.sock')
+            result = receptor_ctl.submit_work(worktype=worktype,
+                                              payload=sockout.makefile('rb'))
+            sockin.close()
+            sockout.close()
+
+            self.update_model(self.instance.pk, receptor_work_id=result['unitid'])
```

Comment:
I was really hoping that receptor would let us specify the work ID, because it should be fairly trivial to give a unique ID with the fields already on the unified job model. Then we wouldn't have to muck with tracking a new field. However...

https://github.com/project-receptor/receptor/blob/f7c245c18911a7ea5edb808bf568c39be21b17a6/receptorctl/receptorctl/socket_interface.py#L148-L151

I realize that the receptorctl code shows no sign of supporting this. Need to cross-reference against the `work submit` CLI itself, and I'm a little rusty to get to its help text right now.

Reply:
cc @ghjm

Reply:
I dont think we want a receptor network being tied to a single Tower installation. If we made this value derived from something like the job id, we're going to run into conflicts.